assets/src/hooks/useDetour.ts

```diff
@@ -13,33 +13,40 @@ import {
 
 const useDetourDirections = (
   shapePoints: ShapePoint[]
-): FetchResult<DetourShape> => {
+): { finishedDirections: boolean; detourShape: FetchResult<DetourShape> } => {
   const [detourShape, setDetourShape] = useState<FetchResult<DetourShape>>(
     loading()
   )
+  const [finishedDirections, setFinishedDirections] = useState(false)
```

Comment:
suggestion: instead of adding a new state, could we derive this status from the `isLoading` function in `fetchResult`?

I think that would mean replacing the `setFinishedDirections(false)` on ln26 to `setDetourShape(loading())`

Reply:
I might be wrong here, but I think that this suggestion will lead to a situation where the directions blink in and out of existence when the user adds or removes waypoints. 

But the existing logic will involve the `Regular Route` instruction disappearing a split second before any other directions when you undo. 

I think the _actual_ thing that's making this hard to think about is the fact that we're doing a lot of this logic in the frontend, when it really belongs in the backend. Let the server decide what the directions are and whether to add in "Regular Route" and whatever else needs to be shown, and let the frontend handle as few separate pieces of state as possible.

To be fair, I don't think we should do that in this PR, or even soon/immediately. But it's starting to look like having detour logic spread between the frontend and the backend is tech debt that's generating higher and higher interest payments.

Reply:
Ah! Ok I was having issues with my envrc that were invisible to me for a while and made me think it was a separate issue. Ok, so I've tried the derived `isLoading` state and saw the blinking directions that Josh noted. I've also seen the lag in `Regular Route` instructions disappearing on undo, as noted. Given the preferred solution may be a refactor, I'll pause on this work and bring that up during stand-up tomorrow so we can weigh as a group whether this tech debt needs doing now or later.